features/login/impl/src/main/kotlin/io/element/android/features/login/impl/accountprovider/AccountProviderDataSource.kt

```diff
@@ -18,21 +19,33 @@ import javax.inject.Inject
 
 @SingleIn(AppScope::class)
 class AccountProviderDataSource @Inject constructor(
-    enterpriseService: EnterpriseService,
+    enterpriseService: EnterpriseService? = null,
 ) {
-    private val defaultAccountProvider = (enterpriseService.defaultHomeserver() ?: AuthenticationConfig.MATRIX_ORG_URL).let { url ->
+    val matrixOrgAccountProvider = AccountProvider(
+        url = AuthenticationConfig.MATRIX_ORG_URL,
+        descriptionResourceId = R.string.screen_change_account_provider_matrix_org_subtitle,
+        isPublic = true,
+    )
+    // add more hard coded AccountProvider here if you need them
+
+    private val defaultAccountProvider = if (enterpriseService?.defaultHomeserver() != null &&
+        enterpriseService.defaultHomeserver() != AuthenticationConfig.MATRIX_ORG_URL) {
         AccountProvider(
-            url = url,
-            subtitle = null,
-            isPublic = url == AuthenticationConfig.MATRIX_ORG_URL,
-            isMatrixOrg = url == AuthenticationConfig.MATRIX_ORG_URL,
+            enterpriseService.defaultHomeserver()!!,
```

Comment:
We try to avoid using `!!` in the codebase. You could use `enterpriseService.defaultHomeserver()?.let { ... }` instead.